app/src/main/java/dev/animeshvarma/sigil/SigilViewModel.kt

```
@@ -203,55 +483,78 @@ class SigilViewModel(application: Application) : AndroidViewModel(application) {
+                        compress = compress,
+                        logCallback = { addLog(it) }
                     )
+                }
+                val wasAutoMode = state.selectedMode == SigilMode.AUTO
+                _uiState.update {
+                    if (wasAutoMode) it.copy(autoOutput = result, isLoading = false)
                     else it.copy(customOutput = result, isLoading = false)
                 }
-            } catch (_: Exception) {
-                addLog("Error: Encryption failed.")
+            } catch (e: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)